### PR TITLE
Round split final (and wrap, merge, unwrap, etc)

### PR DESCRIPTION
**Pull request by tammam1998**
Fixed some history issues with the previous pull request. Contains a working implementation of r_split, r_wrap, r_unwrap, r_merge tested with microbenchmarks.  Also contains some log parsing, testing, and plotting utilities used in benchmarking and comparison to normal auto-split.

### CODE REVIEW

**Comment by angelhof** [comment]
Depends on #161

File: compiler/gather_results.py [hunk 1]
```diff
@@ -51,6 +51,7 @@
 SMALL_UNIX50_RESULTS = "../evaluation/results/unix50_4_1073741824/"
 BIG_UNIX50_RESULTS = "../evaluation/results/unix50_16_10737418240/"
 COARSE_UNIX50_RESULTS = "../evaluation/results/unix50-naive/"
+OASA_RESULTS = "../evaluation/buses/results/"
```
Comment (nvasilakis) [inline compiler/gather_results.py:54]: We might want to start thinking soon about how to have per experiment scripts.

File: docs/eval.md [hunk 2]
```diff
@@ -56,18 +56,18 @@ curl img.pash.ndr.md | docker load; docker run --name pash-playground -it pash/1
...
+* [compiler](../compiler): Shell-Dataflow translations and associated parallelization transformations.
+* [runtime](../runtime): Runtime component — e.g., `eager`, `split`, and associated combiners.
 
 These three components implement the contributions presented in the paper.
 They are expected to be usable with minimal effort, through a few different means presented later---including a `docker` container and log in credentials to our evaluation server.
 
 The auxiliary directories are:
-* [docs](./pash/docs): Design documents, tutorials, installation instructions, etc.
-* [evaluation](./pash/evaluation): Shell pipelines and example [scripts](./pash/evaluation/scripts) used for the evaluation.
+* [docs](../docs): Design documents, tutorials, installation instructions, etc.
```
Comment (nvasilakis) [inline docs/eval.md:67]: Not sure why these are here!

**Review by nvasilakis — APPROVED** [review]
Looks good (and big!:P)

**Comment by nvasilakis** [comment]
Merging this might not be easy, due to conflicts. One way around this would be to make PRs smaller and self-contained:-) (nor sure this helps, I know I'm stating the obvious:P)

**Comment by angelhof** [comment]
I don't think there should be many conflicts with `future`. Merging `future` to main will be a pain though :(

File: compiler/gather_results.py (hunk 1 above)
Comment (angelhof) [reply compiler/gather_results.py:54]: I agree

**Pull request merged**